wifi: Radio test cleanup (#9888)
nrfconnect/sdk-nrf: samples/wifi/radio_test/sample_description.rst

              wifi_radio_test tx 0
              wifi_radio_test tx_pkt_tput_mode 0
              wifi_radio_test tx_pkt_len 4095

>               wifi_radio_test chnl_primary 1

Reviewer: API name to set Channel number is changed from chnl_primary to init. Please correct.

>               wifi_radio_test tx_pkt_rate 6

Reviewer: Before setting wifi_radio_test tx_pkt_rate 6 
set wifi_radio_test tx_pkt_mcs -1

              wifi_radio_test rx 0
              wifi_radio_test tx 0

>               wifi_radio_test init 1

Reviewer 2: Can we use a channel other than 1, which makes this look like an enable flag.

Author: Changed to 11.

              wifi_radio_test rx 0
              wifi_radio_test tx 0

>               wifi_radio_test init 1

Reviewer 2: As above, choose another channel.

Author: Changed to 14.

              * R - Data rate (Mbps)
              * P - PHY overhead duration (us) (Values: 24 us - Legacy OFDM, 192 us - DSSS)

> 

Reviewer 2: Please include an example of using RX mode, and also a follow up showing use of get_stats.

Author: Done

         * To transmit a continuous tone, with the following configuration:

           * Channel: 144
           * TX power: 10 dBm
           * Tone frequency: 2MHz

           Execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test rx 0
              wifi_radio_test tx 0
              wifi_radio_test init 144
              wifi_radio_test tx_power 10
              wifi_radio_test tx_tone_freq 2
              wifi_radio_test tx_tone 1

>          * To stop ongoing transmission of a continuous tone, use the following command:
> 
>               wifi_radio_test tx_tone 0

Reviewer 3: ```suggestion
         * To stop ongoing transmission of a continuous tone, use the following command:
         
           .. code-block:: console

              wifi_radio_test tx_tone 0
```

>          .. note::
> 
>             * wifi_radio_test tx_tone 0 should be only used if the command issued before this is wifi_radio_test tx_tone 1

Reviewer 3: ```suggestion
           .. note::

              Use ``wifi_radio_test tx_tone 0`` only if the command issued before this is ``wifi_radio_test tx_tone 1``.
```

         * To enable DPD hardware block, use the following command:

>               wifi_radio_test dpd 1

Reviewer 2: not rendered as code.

Author: Done.

>          * To enable DPD hardware block, use the following command:
> 
>               wifi_radio_test dpd 1

Reviewer 3: ```suggestion
         * To enable DPD hardware block, use the following command:
         
           .. code-block:: console

              wifi_radio_test dpd 1
```

         * To disable DPD hardware block, use the following command:

>               wifi_radio_test dpd 0

Reviewer 2: not rendered as code.

Author: Done.

>          * To disable DPD hardware block, use the following command:
> 
>               wifi_radio_test dpd 0

Reviewer 3: ```suggestion
         * To disable DPD hardware block, use the following command:
         
           .. code-block:: console

              wifi_radio_test dpd 0
```

         .. note::

            * DPD command can be used only during tone transmission. For TX packet transmission, control of DPD hardware

>             * is taken care by firmware.

Reviewer 2: Rendered incorrectly as two bullets (one is expected).

Author: Done.

>             * DPD command can be used only during tone transmission. For TX packet transmission, control of DPD hardware
>             * is taken care by firmware.

Reviewer 3: ```suggestion
            DPD command can be used only during tone transmission.
            For TX packet transmission, control of DPD hardware is taken care by firmware.
```

         * To read chip temperature in degree Celsius, execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test get_temperature

           The sample shows the following output:

           .. code-block:: console

               wifi_nrf: Temperature reading success:
               wifi_nrf: The temperature is = 21 degree Celsius

         * To measure RF RSSI status, with the following configuration:

           * Channel: 144

           Execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test rx 0
              wifi_radio_test tx 0
              wifi_radio_test init 144
              wifi_radio_test get_rf_rssi

           The sample shows the following output:

           .. code-block:: console

              wifi_nrf: RF RSSI value is = 0

>          .. note::
> 
>             * RF RSSI Status is a 3-bit indicator. Higher the value of RF RSSI, higher is the level of saturation (non-linearity) at the LNA output.

Reviewer 3: ```suggestion
         .. note::

            RF RSSI Status is a 3-bit indicator.
            The higher the value of RF RSSI is, the higher is the level of saturation (non-linearity) at the LNA output.
```

         * To compute optimal XO value, with the following configuration:

           * Channel: 64

           Execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test rx 0
              wifi_radio_test tx 0
              wifi_radio_test init 64
              wifi_radio_test compute_optimal_xo_val

           The sample shows the following output:

           .. code-block:: console

              wifi_nrf: Best XO value is = 43

         .. note::

>             * This test requires a DUT to be connected with VSG using cabled setup. VSG should be continuously generating 0.5MHz tone.
>             * For channel 64, set the VSG frequency to 5320.5MHz. Set signal power level between -40dBm to -50dBm.

Reviewer 3: ```suggestion
            * This test requires a DUT to be connected with VSG using cabled setup.
              VSG should be continuously generating 0.5 MHz tone.
            * For channel 64, set the VSG frequency to 5320.5 MHz.
              Set the signal power level between -40 dBm and -50 dBm.
```

         * To capture ADC samples with fixed RX gain, with the following configuration:

           * Channel: 144
           * LNA gain: 2
           * BB gain: 10
           * Capture length: 64

           Execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test rx 0
              wifi_radio_test tx 0
              wifi_radio_test init 144
              wifi_radio_test rx_lna_gain 2
              wifi_radio_test rx_bb_gain 10
              wifi_radio_test rx_capture_length 64
              wifi_radio_test rx_cap 0

           The sample shows the following output:

           .. code-block:: console

              ************* RX capture data ***********

>               E84FE4

Reviewer 2: These values are not of interest since they will vary from run to run. Can we use some form of <- snip -> notation to only show a few lines ?

Author: Done.

              12805C
              10C0E4
              0C4154
              0541A0
              FD81C8
              F4C1AC
              ED4168
              E880F8
              E64070
              E80FEC
              EC0F78
              F2CF28
              FB8F04
              038F1C
              0B0F60
              104FCC
              12404C
              1180D4
              0D8150
              06819C
              FE01C4
              F601B8
              EE8174
              E90108
              E6C084
              E78FFC
              EB8F84
              F24F28
              FA8F04
              030F14
              0A8F58
              0FCFC0
              124048
              1140CC
              0DC140
              06C19C
              FF01C0
              F6C1B8
              EF017C
              E90110
              E6C08C
              E78004
              EB0F8C
              F18F34
              F9CF04
              024F0C
              0A0F50
              0FCFBC
              124038
              11C0C8
              0D8140
              078194
              FF81C4
              F6C1B4
              EEC178
              E94114

         .. note::

>             * This test requires a DUT to be connected with VSG using cabled setup. VSG should be continuously transmitting
>             * a tone or packets. Set signal power level between -35dBm to -45dBm to get non saturated samples. In the above
>             * example capture is taken for 64 samples. Each complex sample is of 24 bits.
>             * The capture shown above is just an example. The captured samples will vary from run to run

Reviewer 3: ```suggestion
            * This test requires a DUT to be connected with VSG using cabled setup.
              VSG should be continuously transmitting a tone or packets.
            * Set the signal power level between -35 dBm and -45 dBm to get non saturated samples.
              In the example, capture is taken for 64 samples.
              Each complex sample is of 24 bits.
            * The captured samples will vary from run to run.
```

         * To capture baseband samples with fixed RX gain, with the following configuration:

           * Channel: 144
           * LNA gain: 2
           * BB gain: 10
           * Capture length: 64

           Execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test rx 0
              wifi_radio_test tx 0
              wifi_radio_test init 144
              wifi_radio_test rx_lna_gain 2
              wifi_radio_test rx_bb_gain 10
              wifi_radio_test rx_capture_length 64
              wifi_radio_test rx_cap 1

           The sample shows the following output:

           .. code-block:: console

              ************* RX capture data ***********

>               E940F0

Reviewer 2: These values are not of interest since they will vary from run to run. Can we use some form of <- snip -> notation to only show a few lines ?

Author: Done.

              F5519E
              E990F3
              E8FFF4
              F3AF32
              038F20
              0F5FC9
              1040CD
              05D191
              F631A8
              EA1102
              E88FFF
              F2EF36
              029F1D
              0EDFBE
              1090C0
              06718F
              F6A1AB
              EA410B
              E8200B
              F1FF39
              01DF1A
              0E9FBB
              10B0B9
              06D189
              F711AE
              EA6113
              E83015
              F1BF41
              017F17
              0E4FAC
              10B0AB
              076185
              F7A1AD
              EAA11A
              E8101D
              F18F45
              00EF15
              0E0FA7
              10F0A3
              07D180
              F851B4
              EB5120
              E83025
              F0FF4D
              00AF15
              0DDFA1
              1110A2
              08217A
              F871B0
              EB3126
              E79027
              F07F4B
              007F12
              0DAF9B
              115097
              08A178

         .. note::

>             * This test requires a DUT to be connected with VSG using cabled setup. VSG should be continuously transmitting
>             * a tone or packets. Set signal power level between -35dBm to -45dBm to get non saturated samples. In the above
>             * example capture is taken for 64 samples. Each complex sample is of 24 bits.
>             * The capture shown above is just an example. The captured samples will vary from run to run.

Reviewer 3: ```suggestion
            * This test requires a DUT to be connected with VSG using cabled setup.
              VSG should be continuously transmitting a tone or packets.
            * Set the signal power level between -35 dBm and -45 dBm to get non saturated samples.
            *  In the example, capture is taken for 64 samples.
               Each complex sample is of 24 bits.
            * The captured samples will vary from run to run.
```

         * To capture AGC adjusted baseband samples after WLAN packet detection, with the following configuration:

           * Channel: 144
           * Capture length: 64

           Execute the following sequence of commands:

           .. code-block:: console

              wifi_radio_test rx 0
              wifi_radio_test tx 0
              wifi_radio_test init 144
              wifi_radio_test rx_capture_length 64
              wifi_radio_test rx_dyn_pkt_cap

           The sample shows the following output:

           .. code-block:: console

              ************* RX capture data ***********

>               83E6824F

Reviewer 2: These values are not of interest since they will vary from run to run. Can we use some form of <- snip -> notation to only show a few lines ?

Author: Done.

              50BCE2D2
              0A1D63E1
              1710F0AD
              0DDFE040
              25B128F4
              C51DF1AB
              B0242261
              E4EDD719
              83EE40FF
              4DD4181F
              DC31DBFE
              5BE3E22B
              D09CE042
              0CED70E0
              001050BE
              2DD4E070
              24C149F6
              DF1CF1C2
              C03A2121
              E69DC619
              90EDF0F4
              2DD0184F
              DAE1CEFF
              63E2420C
              D059E002
              124DA3DE
              010B9054
              9F83000F
              0FFE3030
              1703FDBF
              C0420902
              D9DE06E0
              F0D41135
              9E18DC10
              0ACFF9FE
              C82590E0
              CE0C041D
              19FF21EA
              FB0A10E7
              6EA00C92
              0FCF7A24
              2DFCB276
              5100DA9F
              FDE1B9D6
              4E12A177
              6F64EB52
              EC1187CD

         .. note::

>             * This test requires a DUT to be connected with VSG using cabled setup. VSG should be continuously transmitting
>             * WLAN packets. Set the desired signal power.
>             * The above capture is taken for 64 samples. Each complex sample is of 24 bits. The console will display in 32 bits
>             * format so there are 48, 32 bit samples.
>             * In the above capture 0xE6824F is first sample, 0xE2D283 is second sample, 0xE150BC is third sample and so on
>             * The capture shown above is just an example. The captured samples will vary from run to run
>             * The above capture is taken after WLAN packet detection. So, this capture will not have first few samples in
>             * first WLAN packet in the capture.
>             * The user is advised to feed smaller packets so that multiple packets can be seen in the capture.

Reviewer 3: ```suggestion
            * This test requires a DUT to be connected with VSG using cabled setup.
              VSG should be continuously transmitting WLAN packets.
            * Set the desired signal power.
            * The capture is taken for 64 samples.
              Each complex sample is of 24 bits.
              The console will display in 32 bits format so there are 48, 32 bit samples.
            * In the capture, ``0xE6824F`` is the first sample, ``0xE2D283`` the second sample, ``0xE150BC`` the third sample and so on.
            * The capture is just an example.
              The captured samples will vary from run to run
            * The capture is taken after WLAN packet detection, so it will not have the first few samples in the first WLAN packet.
            * Smaller packets should be used so that multiple packets can be seen in the capture.
```
